src/libs/installer/installereventoperation.cpp

```diff
@@ -0,0 +1,347 @@
+#include "installereventoperation.h"
+
+#include "eventlogger.h"
+#include "packagemanagercore.h"
+#include "globals.h"
+
+#include <QtCore/QDebug>
+
+using namespace QInstaller;
+
+InstallerEventOperation::InstallerEventOperation(PackageManagerCore *core)
+    : UpdateOperation(core)
+{
+    setName(QLatin1String("InstallerEvent"));
+}
+
+void InstallerEventOperation::backup()
+{
+}
+
+bool InstallerEventOperation::performOperation()
+{
+    // todo: Make sure we have the correct number of arguments
+
+    const QStringList args = arguments();
+    const QString action = args.at(0);
+
+    if(action == QString::fromLatin1("init"))
+    {
+        return sendInit(args);
+    }
+    else if(action == QString::fromLatin1("uninstaller"))
+    {
+        return sendUninstallerEvent(args);
+    }
+    else if(action == QString::fromLatin1("installer"))
+    {
+        return sendInstallerEvent(args);
+    }
+
+    return false;
+}
+
+// init takes: the following parameters
+//   1. region: "world" | "china"
+//   2. version: "1.0.x"
+//   3. environment: "release" | "dev"
+//   4. provider: "none" | "A" | "B" | ...
+bool InstallerEventOperation::sendInit(QStringList args)
+{
+    // if (!checkArgumentCount(5, 5, tr(" ([\"init\"|message type], [\"china\"|\"world\"] (init only), version (init only), [\"release\"|\"dev\"] (init only), [\"none\"|\"A\"|...] (init only))")))
+    //     return false;
+
+    // Get region
+    eve_launcher::application::Application_Region region = eve_launcher::application::Application_Region_REGION_WORLD;
+    if (args.at(1) == QString::fromLatin1("china"))
+    {
+        region = eve_launcher::application::Application_Region_REGION_CHINA;
+    }
+
+    // Get version
+    QString version = args.at(2);
+
+    // Get build type
+    eve_launcher::application::Application_BuildType buildType = eve_launcher::application::Application_BuildType_BUILDTYPE_DEV;
+    if (args.at(3) == QString::fromLatin1("release"))
+    {
+        buildType = eve_launcher::application::Application_BuildType_BUILDTYPE_RELEASE;
+    }
+
+    // Get provider
+    bool provider = false;
+    QString providerName = QString::fromLatin1("");
+    if (args.at(4) != QString::fromLatin1("none"))
+    {
+        provider = true;
+        providerName = args.at(4);
+    }
+
+    EventLogger::instance()->initialize(region, version, buildType, provider, providerName);
+
+    return true;
+}
+
+bool InstallerEventOperation::sendUninstallerEvent(QStringList args)
+{
+    const QString event = args.at(1);
+    
+    if (event == QString::fromLatin1("Started")) {
```

Comment:
Could you put the bracket in the next line

Reply:
Old habits, so used to having the bracket start in the same line (since that is the js standard)